crates/sui-indexer-alt-framework/src/mocks/store.rs

```diff
@@ -441,28 +504,33 @@ impl MockStore {
     }
 
     /// Wait for any data to be processed and stored, panicking if timeout is reached
-    pub async fn wait_for_any_data(&self, timeout_duration: Duration) {
+    pub async fn wait_for_any_data(&self, watermark_key: &str, timeout_duration: Duration) {
         let start = std::time::Instant::now();
         while start.elapsed() < timeout_duration {
             {
-                let data = self.data.lock().unwrap();
-                if !data.is_empty() {
+                if self.data.contains_key(watermark_key) {
                     return;
                 }
             }
-            tokio::time::sleep(Duration::from_millis(50)).await;
+            tokio::time::sleep(Duration::from_millis(1000)).await;
```

Comment:
1000 millis (1 second) might be too long here. We have some tests whose `timeout_duration` is 1 second, which means we only get one attempt at getting the data. Maybe use 100 millis instead?